### ISSUE TITLE
Training does not converg

### ISSUE THREAD

**Issue by luben2018** [zouchuhang/LayoutNet#18]
hello,
thank you for sharing the great work. 
I tried the edge training 'th driver_pano_edg.lua' followed with your guidance, but the loss value does not converge. I checked the code and the data but can not find the reason. could you give me some advices?

the output as followed

done	
414	
Uploaded training	
46	
Uploaded validation	
start training	
update param, loss = 0.64732569, gradnorm = 6.7421e+00	
update param, loss = 2.26082158, gradnorm = 9.6880e-01	
update param, loss = 2.10377669, gradnorm = 1.5707e-03	
update param, loss = 2.14185762, gradnorm = 0.0000e+00	
update param, loss = 2.16074014, gradnorm = 0.0000e+00	
update param, loss = 2.17253232, gradnorm = 0.0000e+00	
...
update param, loss = 2.05997038, gradnorm = 0.0000e+00	
update param, loss = 2.17931867, gradnorm = 0.0000e+00	
update param, loss = 2.07385349, gradnorm = 0.0000e+00	
iteration 8000, loss = 2.07385349, gradnorm = 0.0000e+00	
validation loss = 2.14375149

**Comment by zouchuhang** [comment]
@luben2018 The printed "gradnorm" shows zero gradients. I guess your training get stuck, rerun the code is the simplest strategy to solve.

**Comment by luben2018** [comment]
@zouchuhang thanks for the kind reply!
I tried several times, but all failed. so this is normal? do you have the same problem during the training? any tips? or I just have to try more?  during the successful training of the edges, what is the value of the loss? 
looking forward to your further advices! and thanks again!

**Comment by luben2018** [comment]
@zouchuhang thanks for the kind reply. 
it is funny that the training begins to converge when I am about to complain after dozen attempts.  so, happy, and thanks sincerely.

**Comment by zouchuhang** [comment]
@luben2018 Glad to see it works:)
